### ISSUE TITLE
Very long compilation times for llama2 with batch size 4

### ISSUE THREAD

**Issue by dacorvo** [aws-neuron/transformers-neuronx#48]
With AWS Neuron SDK `2.14.1`, I am experiencing very long compilation times for `batch_size = 4` with the [llama2 7B](https://huggingface.co/meta-llama/Llama-2-7b-hf) model.

I am using the following configurations:

```
|             | inf2.8xlarge | inf2.48xlarge |
|-------------|--------------|---------------|
| tp_degree   | 2            | 24            |
| n_positions | 2048         | 2048          |
| amp         | f16          | f16           |
```

With `batch_size = 1, 2` it takes minutes to compile the model with the `-O1` option, but with `batch_size = 4` it lasts more than three hours.

**Comment by awsilya** [comment]
ack, asking compiler team to take a look

**Comment by aws-donkrets** [comment]
Hi [dacorvo](https://github.com/dacorvo)  - We have root caused this issue with the llama2 batch 4 config. The fix will be in the next Neuron SDK release.

**Comment by dacorvo** [comment]
This is excellent news ! Thanks for the update.

**Comment by jeffhataws** [comment]
This issue was fixed in Neuron SDK release 2.15.
